hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleContextImpl.java

```diff
@@ -607,6 +609,10 @@ private void dispatchSyntheticTxn(
             dispatcher.dispatchPureChecks(txBody);
         } catch (PreCheckException e) {
             childRecordBuilder.status(e.responseCode());
+            var output = ReturnTypes.encodedRc(e.responseCode());
```

Comment:
It seems to me like we are making an assumption here that all PreCheckExceptions result from a contract call. I'm not sure if this is true and perhaps this can lead to unexpected outcomes.